pentaho-js/src/main/javascript/cdf/components/TableComponent.js

```diff
@@ -679,16 +680,16 @@ define([
       tablePlaceholder.trigger('cdfTableComponentFinishRendering');
     },
 
-    __getDataTableData: function(chartDefinition) {
-      var dtData0 = TableComponent.getDataTableOptions(chartDefinition);
+    __getDataTableOptions: function(chartDefinition) {
+      var dataTableOptions = TableComponent.getDataTableOptions(chartDefinition);
 
       // Build a default config from the standard options
       var extraOptions = {};
       $.each(this.extraOptions != null ? this.extraOptions : {}, function(index, option) {
         extraOptions[option[0]] = option[1];
       });
 
-      return $.extend(chartDefinition.dataTableOptions, dtData0, extraOptions);
+      return $.extend({}, chartDefinition.dataTableOptions, dataTableOptions, extraOptions);
```

Comment:
I also pondered on complaining about this mutating `chartDefinition.dataTableOptions`, but again, to preserve backward compatibility, I think we should keep it the way it was. Nice catch though!